src/main/java/org/phyloref/jphyloref/commands/TestCommand.java

```diff
@@ -258,25 +303,101 @@ public void execute(CommandLine cmdLine) throws RuntimeException {
+                                        hasIntervalEndDate = true;
+                                    }
+                                }
+                            }
+                        }
+
+                        else if(axiom.getProperty().equals(pso_withStatus)) {
+                            currentStatuses.add((IRI)axiom.getValue());
+                        } else {
+                            System.err.println("Unknown axiom: " + axiom);
+                            System.exit(-1);
+                        }
+                    }
+            	}
+
+                // Did the current statuses have a start date but no end date, i.e. are they currently active?
+                if(hasIntervalStartDate && !hasIntervalEndDate) {
+                    statuses.addAll(currentStatuses);
+                }
+            }
+
+            // System.err.println("Active statuses: " + statuses);
+
+            // Based on the phyloreference status, we can determine whether or not
+            // we expect the phyloreference to actually resolve: we do if there
+            // were no statuses, or if the statuses include submitted or published.
+            // In all other cases, we're not expecting the phyloreference to resolve.
+            boolean flag_expected_to_resolve = (
+                statuses.isEmpty() ||
+                statuses.contains(IRI.create("http://purl.org/spar/pso/submitted")) ||
+                statuses.contains(IRI.create("http://purl.org/spar/pso/published"))
+            );
+
             // Determine if this phyloreference has failed or succeeded.
             if(testFailed) {
             	if(unmatched_specifiers.isEmpty()) {
-	                // Yay, failure!
-	                countFailure++;
-	                result.setStatus(StatusValues.NOT_OK);
-	                testSet.addTapLine(result);
-            	} else {
-            		// Okay, it's a failure, but we do know that there are unmatched specifiers.
-            		// So mark it as a to-do.
-            		countTODO++;
+                    if(flag_expected_to_resolve) {
+                        // We expect this phyloreference to resolve. Report a failure.
+                        countFailure++;
+                        result.setStatus(StatusValues.NOT_OK);
+                        testSet.addTapLine(result);
+                    } else {
+                        // No, we do not expect this phyloreference to resolve. Report a TODO.
+                        countTODO++;
             		result.setStatus(StatusValues.NOT_OK);
-            		result.setDirective(new Directive(DirectiveValues.TODO, "Phyloreference could not be tested, as one or more specifiers did not match."));
+            		result.setDirective(new Directive(DirectiveValues.TODO, "Phyloreference did not resolve, but has status " + statuses));
             		testSet.addTapLine(result);
```

Comment:
Seems like the indentation of this and the preceding two lines is off by one space?